mysite/polls/views.py

```diff
@@ -1,43 +1,39 @@
-from django.shortcuts import render
-import search
+import re
+
 import json
+from django.shortcuts import render
 from django.http import JsonResponse
 from django.http import HttpResponse
-import re
-# Create your views here.
-
 
-def make_link_for_html(movie_id):
-    link = '<a href = "http://localhost:8000/polls/info/?q=' \
-            + movie_id + '"> Link </a>'
-    return link
+import search
+import imdb_url
+# Create your views here.
 
 
 def make_link(movie_id):
-    link = 'http://localhost:8000/polls/info/?q=' \
-            + movie_id  
+    link = "%s%s" %(imdb_url.query_url, movie_id) 
     return link
 
 
 def index(request):
-    ans = request.GET.get('q', 'Nothing')
-    movie_ids, movie_names, id_name_map = search.get_movie_names(ans)
-    print movie_ids
-    answer = ""
+    ans = request.GET.get('q', None)
+    if ans is None or ans ==  "":
+        return HttpResponse("Information request cannot be fulfilled")
+    movie_ids, movie_names = search.get_movie_names(ans)
+    if movie_ids == []:
+        return HttpResponse("No Movies could be Found")
     json_answer = {}
-    for movie_index in range(0, len(movie_ids)):
-        link = make_link_for_html(movie_ids[movie_index])
-        link2 = make_link(movie_ids[movie_index])
-        ans = link + " " + movie_names[movie_index] + "<br>"
-        answer += link + " " + movie_names[movie_index] + "<br>"
-        json_answer[movie_names[movie_index]] = link2
-    print "\n"
+    movie_tuple = zip(movie_ids, movie_names)
+    for tup in movie_tuple:
+        link = make_link(tup[0])
+        json_answer[tup[1]] = link
     return JsonResponse(json_answer) 
-    return HttpResponse(answer)
 
 
 def display_info(request):
-    ans = request.GET.get('q', 'Nothing')
+    ans = request.GET.get('q', None)
+    if ans is None or ans ==  "":
+        return HttpResponse("Information request cannot be fulfilled")
```

Comment:
if your API returns JSON response then you should return your errors too in the json